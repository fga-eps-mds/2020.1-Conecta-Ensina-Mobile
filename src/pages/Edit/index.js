import React, {useContext} from 'react';
import Theme from '../../../Theme';
import {Formik} from 'formik';
import Background1 from '../../components/Background1';
import CustomText from '../../components/CustomText';
import EditField from '../../components/EditField';
import GeneralPicker from '../../components/GeneralPicker';
import * as yup from 'yup';

import {AuthContext} from '../../contexts/auth';

import {
  TextContainer,
  Container,
  InfoContainer,
  ButtonContainer,
  ContainerBotao,
} from './styles';

export default function Edit({navigation}) {
  const {user, student, updateUser} = useContext(AuthContext);
  function handleEdit(values) {
    console.log('passei aqui');
    updateUser(values, user.id);
    navigation.navigate('Profile');
  }
  let initials = {
    name: user && user.firstName,
    surname: user && user.lastName,
    email: user && user.email,
    cellphone: user && user.cellphone,
    grade: student && student.grade,
    school: student && student.institution,
    cep: student && student.cep,
    num: student && student.number,
    details: student && student.details ? student.details : '',
    description: student && student.description ? student.description : '',
  };

  let Schema = yup.object().shape({
    name: yup
      .string('Nome deve ser um texto')
      .required('É necessário indicar um nome'),
    surname: yup
      .string('Sobrenome deve ser um texto')
      .required('É necessário indicar um sobrenome'),
    email: yup
      .string('Email deve ser um texto')
      .email('Email deve ter formato válido')
      .required('É necessário indicar um email'),
    cellphone: yup
      .string()
      .required('É necessário indicar um número de telefone')
      .min(11, 'Número de telefone inválido')
      .max(13, 'Número de telefone inválido'),
    grade: yup.number().required('É necessário indicar uma série'),
    school: yup
      .string('Instituição deve ser um texto')
      .required('É necessário indicar uma instituição'),
    cep: yup
      .string()
      .length(8, 'CPF deve ter 8 números')
      .matches(/^\d+$/, 'CEP deve ser um número')
      .required('É necessário indicar um CEP'),
    num: yup
      .number('Número do endereço deve ser um numero')
      .required('É necessário indicar um número do endereço'),
    details: yup.string('Complemento deve ser um texto').nullable(),
    description: yup.string().nullable(),
  });
  let info;

  return (
    <Theme>
      <Background1>
        <Formik
          enableReinitialize
          initialValues={initials}
          validateOnChange={false}
          validateOnBlur={false}
          onSubmit={handleEdit}
          validationSchema={Schema}>
          {({
            handleChange,
            handleSubmit,
            setFieldValue,
            values,
            errors,
            touched,
          }) => (
            <>
              <Container>
                <InfoContainer>
                  {
                    ((info = [
                      {
                        field: 'Nome',
                        value: values.name,
                        change: 'name',
                        error: errors && errors.name,
                      },
                      {
                        field: 'Sobrenome',
                        value: values.surname,
                        change: 'surname',
                        error: errors && errors.surname,
                      },
                      {
                        field: 'Email',
                        value: values.email,
                        change: 'email',
                        error: errors && errors.email,
                      },
                      {
                        field: 'Celular',
                        value: values.cellphone,
                        change: 'cellphone',
                        error: errors && errors.cellphone,
                      },
                      {
                        field: 'CEP',
                        value: values.cep,
                        change: 'cep',
                        error: errors && errors.cep,
                      },
                      {
                        field: 'Número do endereço',
                        value: values.num,
                        change: 'num',
                        error: errors && errors.num,
                        placeholder: values.num.toString(),
                      },
                      {
                        field: 'Complemento do endereço',
                        value: values.details,
                        change: 'details',
                        error: errors && errors.details,
                      },
                      {
                        field: 'Descrição',
                        value: values.description,
                        change: 'description',
                        error: errors && errors.description,
                      },
                      {
                        field: 'Instituição',
                        value: values.institution,
                        change: 'institution',
                        error: errors && errors.institution,
                      },
                    ]),
                    info.map((item, index) => {
                      return (
                        <EditField
                          key={index}
                          field={item.field}
                          placeholder={item.placeholder}
                          value={item.value}
                          onChangeText={handleChange(item.change)}
                          errors={item.error}
                        />
                      );
                    }))
                  }
                  <CustomText black medium>
                    Série
                  </CustomText>
                  <TextContainer>
                    <GeneralPicker
                      grade
                      placeholderValue={student && student.grade}
                      onChange={(value) => setFieldValue('grade', value, false)}
                    />
                    {errors.grade && (
                      <CustomText black small>
                        {errors.grade}
                      </CustomText>
                    )}
                  </TextContainer>
                </InfoContainer>
              </Container>
              <ContainerBotao>
                <ButtonContainer onPress={handleSubmit} testID="button">
                  <CustomText white bigSmall>
                    Salvar
                  </CustomText>
                </ButtonContainer>
              </ContainerBotao>
            </>
          )}
        </Formik>
      </Background1>
    </Theme>
  );
}
/*(info = [
                          {
                            text:
                              user && user.firstName
                                ? user.firstName
                                : 'Não informado',
                            field: 'Nome',
                            value: values.name,
                            change: 'name',
                            error: errors && errors.name,
                          },
                        ]),*/
