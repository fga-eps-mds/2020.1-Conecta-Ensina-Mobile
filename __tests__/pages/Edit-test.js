import React from 'react';
import {render, fireEvent, act} from '@testing-library/react-native';
import Edit from '../../src/pages/Edit';

describe('Testing Edit page', () => {
  test('Should renders page', () => {
    const tree = render(<Edit />).toJSON();

    expect(tree).toMatchSnapshot();
  });
  test("Should press 'Submit' button", async () => {
    const {getByTestId, debug} = render(<Edit />);

    const button = getByTestId('button');

    await act(async () => {
      await fireEvent.press(button);
    });
    //debug();
  });
});
