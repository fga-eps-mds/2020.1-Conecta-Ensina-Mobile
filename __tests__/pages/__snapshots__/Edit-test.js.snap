// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Edit page Should renders page 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#184E79",
        "display": "flex",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "flex-end",
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexDirection": "row-reverse",
        "flexGrow": 10,
        "flexShrink": 1,
        "marginBottom": "5%",
        "opacity": 1,
        "paddingBottom": "10%",
        "paddingLeft": "10%",
        "paddingRight": "10%",
        "paddingTop": "10%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#ffffff",
            "fontFamily": "Roboto",
            "fontSize": 20,
            "textAlign": "center",
          },
        ]
      }
      white={true}
    >
      Selecione as disciplinas em que deseja lecionar
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "borderTopLeftRadius": 20,
          "borderTopRightRadius": 20,
          "display": "flex",
          "flexBasis": 0,
          "flexGrow": 78,
          "flexShrink": 1,
          "justifyContent": "center",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "borderColor": "#184E79",
            "borderRadius": 20,
            "borderStyle": "solid",
            "borderWidth": 4,
            "bottom": "13.3%",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": "-5.7%",
          },
        ]
      }
    >
      <RCTScrollView
        contentContainerStyle={
          Object {
            "alignItems": "center",
          }
        }
        style={
          Array [
            Object {
              "height": "64.28%",
              "marginTop": "7.6%",
              "width": "100%",
            },
          ]
        }
      >
        <View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#184E79",
                  "borderRadius": 9,
                  "height": 50,
                  "justifyContent": "center",
                  "marginTop": 20,
                  "width": 280,
                },
              ]
            }
          >
            <TextInput
              allowFontScaling={true}
              multiline={true}
              onChangeText={[Function]}
              placeholderTextColor="#F6F6F6"
              rejectResponderTermination={true}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#184E79",
                    "borderRadius": 10,
                    "color": "#ffffff",
                    "height": 45,
                    "justifyContent": "center",
                    "marginTop": "5%",
                    "width": 280,
                  },
                ]
              }
              textAlign="center"
              underlineColorAndroid="transparent"
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#184E79",
                  "borderRadius": 9,
                  "height": 50,
                  "justifyContent": "center",
                  "marginTop": 20,
                  "width": 280,
                },
              ]
            }
          >
            <TextInput
              allowFontScaling={true}
              multiline={true}
              onChangeText={[Function]}
              placeholderTextColor="#F6F6F6"
              rejectResponderTermination={true}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#184E79",
                    "borderRadius": 10,
                    "color": "#ffffff",
                    "height": 45,
                    "justifyContent": "center",
                    "marginTop": "5%",
                    "width": 280,
                  },
                ]
              }
              textAlign="center"
              underlineColorAndroid="transparent"
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#184E79",
                  "borderRadius": 9,
                  "height": 50,
                  "justifyContent": "center",
                  "marginTop": 20,
                  "width": 280,
                },
              ]
            }
          >
            <TextInput
              allowFontScaling={true}
              multiline={true}
              onChangeText={[Function]}
              placeholderTextColor="#F6F6F6"
              rejectResponderTermination={true}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#184E79",
                    "borderRadius": 10,
                    "color": "#ffffff",
                    "height": 45,
                    "justifyContent": "center",
                    "marginTop": "5%",
                    "width": 280,
                  },
                ]
              }
              textAlign="center"
              underlineColorAndroid="transparent"
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#184E79",
                  "borderRadius": 9,
                  "height": 50,
                  "justifyContent": "center",
                  "marginTop": 20,
                  "width": 280,
                },
              ]
            }
          >
            <TextInput
              allowFontScaling={true}
              multiline={true}
              onChangeText={[Function]}
              placeholderTextColor="#F6F6F6"
              rejectResponderTermination={true}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#184E79",
                    "borderRadius": 10,
                    "color": "#ffffff",
                    "height": 45,
                    "justifyContent": "center",
                    "marginTop": "5%",
                    "width": 280,
                  },
                ]
              }
              textAlign="center"
              underlineColorAndroid="transparent"
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#184E79",
                  "borderRadius": 9,
                  "height": 50,
                  "justifyContent": "center",
                  "marginTop": 20,
                  "width": 280,
                },
              ]
            }
          >
            <TextInput
              allowFontScaling={true}
              multiline={true}
              onChangeText={[Function]}
              placeholderTextColor="#F6F6F6"
              rejectResponderTermination={true}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#184E79",
                    "borderRadius": 10,
                    "color": "#ffffff",
                    "height": 45,
                    "justifyContent": "center",
                    "marginTop": "5%",
                    "width": 280,
                  },
                ]
              }
              textAlign="center"
              underlineColorAndroid="transparent"
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#184E79",
                  "borderRadius": 9,
                  "height": 50,
                  "justifyContent": "center",
                  "marginTop": 20,
                  "width": 280,
                },
              ]
            }
          >
            <TextInput
              allowFontScaling={true}
              multiline={true}
              onChangeText={[Function]}
              placeholderTextColor="#F6F6F6"
              rejectResponderTermination={true}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#184E79",
                    "borderRadius": 10,
                    "color": "#ffffff",
                    "height": 45,
                    "justifyContent": "center",
                    "marginTop": "5%",
                    "width": 280,
                  },
                ]
              }
              textAlign="center"
              underlineColorAndroid="transparent"
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#184E79",
                  "borderRadius": 9,
                  "height": 50,
                  "justifyContent": "center",
                  "marginTop": 20,
                  "width": 280,
                },
              ]
            }
          >
            <TextInput
              allowFontScaling={true}
              multiline={true}
              onChangeText={[Function]}
              placeholderTextColor="#F6F6F6"
              rejectResponderTermination={true}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#184E79",
                    "borderRadius": 10,
                    "color": "#ffffff",
                    "height": 45,
                    "justifyContent": "center",
                    "marginTop": "5%",
                    "width": 280,
                  },
                ]
              }
              textAlign="center"
              underlineColorAndroid="transparent"
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#184E79",
                  "borderRadius": 9,
                  "height": 50,
                  "justifyContent": "center",
                  "marginTop": 20,
                  "width": 280,
                },
              ]
            }
          >
            <TextInput
              allowFontScaling={true}
              multiline={true}
              onChangeText={[Function]}
              placeholderTextColor="#F6F6F6"
              rejectResponderTermination={true}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#184E79",
                    "borderRadius": 10,
                    "color": "#ffffff",
                    "height": 45,
                    "justifyContent": "center",
                    "marginTop": "5%",
                    "width": 280,
                  },
                ]
              }
              textAlign="center"
              underlineColorAndroid="transparent"
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#184E79",
                  "borderRadius": 9,
                  "height": 50,
                  "justifyContent": "center",
                  "marginTop": 20,
                  "width": 280,
                },
              ]
            }
          >
            <TextInput
              allowFontScaling={true}
              multiline={true}
              onChangeText={[Function]}
              placeholderTextColor="#F6F6F6"
              rejectResponderTermination={true}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#184E79",
                    "borderRadius": 10,
                    "color": "#ffffff",
                    "height": 45,
                    "justifyContent": "center",
                    "marginTop": "5%",
                    "width": 280,
                  },
                ]
              }
              textAlign="center"
              underlineColorAndroid="transparent"
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#184E79",
                  "borderRadius": 9,
                  "height": 50,
                  "justifyContent": "center",
                  "marginTop": 20,
                  "width": 280,
                },
              ]
            }
          >
            <View
              grade={true}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "borderRadius": 10,
                    "width": 71,
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#184E79",
                      "borderRadius": 10,
                      "height": 45,
                      "width": "100%",
                    },
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignSelf": "stretch",
                      },
                      undefined,
                    ]
                  }
                >
                  <View
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                    testID="ios_touchable_wrapper"
                  >
                    <View
                      pointerEvents="box-only"
                    >
                      <TextInput
                        allowFontScaling={true}
                        editable={false}
                        rejectResponderTermination={true}
                        style={
                          Array [
                            undefined,
                            Object {
                              "color": "#ffffff",
                              "fontSize": 12,
                            },
                          ]
                        }
                        testID="text_input"
                        underlineColorAndroid="transparent"
                        value="Serie"
                      />
                    </View>
                  </View>
                  <Modal
                    hardwareAccelerated={false}
                    onOrientationChange={[Function]}
                    supportedOrientations={
                      Array [
                        "portrait",
                        "landscape",
                      ]
                    }
                    testID="ios_modal"
                    transparent={true}
                    visible={false}
                  >
                    <View
                      accessible={true}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "flex": 1,
                          "opacity": 1,
                        }
                      }
                      testID="ios_modal_top"
                    />
                    <View
                      style={
                        Array [
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "#f8f8f8",
                            "borderTopColor": "#dedede",
                            "borderTopWidth": 1,
                            "flexDirection": "row",
                            "height": 45,
                            "justifyContent": "space-between",
                            "paddingHorizontal": 10,
                            "zIndex": 2,
                          },
                          undefined,
                        ]
                      }
                      testID="input_accessory_view"
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "flexDirection": "row",
                            },
                            undefined,
                          ]
                        }
                      >
                        <View
                          accessible={true}
                          focusable={true}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "opacity": 1,
                            }
                          }
                        >
                          <View
                            style={
                              Array [
                                Object {
                                  "backgroundColor": "transparent",
                                  "borderColor": "#a1a1a1",
                                  "borderRightWidth": 1.5,
                                  "borderTopWidth": 1.5,
                                  "height": 15,
                                  "width": 15,
                                },
                                undefined,
                                Object {
                                  "marginLeft": 11,
                                  "transform": Array [
                                    Object {
                                      "translateY": 4,
                                    },
                                    Object {
                                      "rotate": "-45deg",
                                    },
                                  ],
                                },
                                undefined,
                                Object {},
                              ]
                            }
                          />
                        </View>
                        <View
                          accessible={true}
                          focusable={true}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "opacity": 1,
                            }
                          }
                        >
                          <View
                            style={
                              Array [
                                Object {
                                  "backgroundColor": "transparent",
                                  "borderColor": "#a1a1a1",
                                  "borderRightWidth": 1.5,
                                  "borderTopWidth": 1.5,
                                  "height": 15,
                                  "width": 15,
                                },
                                undefined,
                                Object {
                                  "marginLeft": 22,
                                  "transform": Array [
                                    Object {
                                      "translateY": -5,
                                    },
                                    Object {
                                      "rotate": "135deg",
                                    },
                                  ],
                                },
                                undefined,
                                Object {},
                              ]
                            }
                          />
                        </View>
                      </View>
                      <View
                        accessible={true}
                        focusable={true}
                        hitSlop={
                          Object {
                            "bottom": 4,
                            "left": 4,
                            "right": 4,
                            "top": 4,
                          }
                        }
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "opacity": 1,
                          }
                        }
                        testID="done_button"
                      >
                        <View
                          testID="needed_for_touchable"
                        >
                          <Text
                            allowFontScaling={false}
                            style={
                              Array [
                                Object {
                                  "color": "#007aff",
                                  "fontSize": 17,
                                  "fontWeight": "600",
                                  "paddingRight": 11,
                                  "paddingTop": 1,
                                },
                                undefined,
                                Object {},
                              ]
                            }
                            testID="done_text"
                          >
                            Done
                          </Text>
                        </View>
                      </View>
                    </View>
                    <View
                      style={
                        Array [
                          Object {
                            "backgroundColor": "#d0d4da",
                            "justifyContent": "center",
                          },
                          Object {
                            "height": 215,
                          },
                          undefined,
                        ]
                      }
                    >
                      <View>
                        <RNCPicker
                          items={
                            Array [
                              Object {
                                "label": "Serie",
                                "textColor": 4278190080,
                                "value": null,
                              },
                              Object {
                                "label": "Educação Infantil",
                                "textColor": 4278190080,
                                "value": "0",
                              },
                              Object {
                                "label": "1º Ano do Ensino Fundamental",
                                "textColor": 4278190080,
                                "value": "1",
                              },
                              Object {
                                "label": "2º Ano do Ensino Fundamental",
                                "textColor": 4278190080,
                                "value": "2",
                              },
                              Object {
                                "label": "3º Ano do Ensino Fundamental",
                                "textColor": 4278190080,
                                "value": "3",
                              },
                              Object {
                                "label": "4º Ano do Ensino Fundamental",
                                "textColor": 4278190080,
                                "value": "4",
                              },
                              Object {
                                "label": "5º Ano do Ensino Fundamental",
                                "textColor": 4278190080,
                                "value": "5",
                              },
                              Object {
                                "label": "6º Ano do Ensino Fundamental",
                                "textColor": 4278190080,
                                "value": "6",
                              },
                              Object {
                                "label": "7º Ano do Ensino Fundamental",
                                "textColor": 4278190080,
                                "value": "7",
                              },
                              Object {
                                "label": "8º Ano do Ensino Fundamental",
                                "textColor": 4278190080,
                                "value": "8",
                              },
                              Object {
                                "label": "9º Ano do Ensino Fundamental",
                                "textColor": 4278190080,
                                "value": "9",
                              },
                              Object {
                                "label": "Ensino Fundamental Completo",
                                "textColor": 4278190080,
                                "value": "10",
                              },
                              Object {
                                "label": "1º Ano do Ensino Médio",
                                "textColor": 4278190080,
                                "value": "11",
                              },
                              Object {
                                "label": "2º Ano do Ensino Médio",
                                "textColor": 4278190080,
                                "value": "12",
                              },
                              Object {
                                "label": "3º Ano do Ensino Médio",
                                "textColor": 4278190080,
                                "value": "13",
                              },
                              Object {
                                "label": "Ensino Médio Completo",
                                "textColor": 4278190080,
                                "value": "14",
                              },
                              Object {
                                "label": "Ensino Superior Incomplento",
                                "textColor": 4278190080,
                                "value": "15",
                              },
                              Object {
                                "label": "Bacharelado/Licensiatura Completo",
                                "textColor": 4278190080,
                                "value": "16",
                              },
                              Object {
                                "label": "Pós-graduação Incompleta",
                                "textColor": 4278190080,
                                "value": "17",
                              },
                              Object {
                                "label": "Pós-graduação completa",
                                "textColor": 4278190080,
                                "value": "18",
                              },
                              Object {
                                "label": "Mestrado Incompleto",
                                "textColor": 4278190080,
                                "value": "19",
                              },
                              Object {
                                "label": "Mestrado",
                                "textColor": 4278190080,
                                "value": "20",
                              },
                            ]
                          }
                          onChange={[Function]}
                          selectedIndex={0}
                          style={
                            Array [
                              Object {
                                "height": 216,
                              },
                              undefined,
                            ]
                          }
                          testID="ios_picker"
                        />
                      </View>
                    </View>
                  </Modal>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#E46270",
          "borderRadius": 5,
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "height": 50,
          "justifyContent": "center",
          "marginTop": "110%",
          "opacity": 1,
          "width": 280,
        }
      }
      testID="button"
    >
      <Text
        bigSmall={true}
        style={
          Array [
            Object {
              "color": "#ffffff",
              "fontFamily": "Roboto",
              "fontSize": 16,
              "textAlign": "center",
            },
          ]
        }
        white={true}
      >
        Salvar
      </Text>
    </View>
  </View>
</View>
`;
